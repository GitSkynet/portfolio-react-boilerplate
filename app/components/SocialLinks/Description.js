import styled from 'styled-components';

const Description = styled.span`
  display: inherit;
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  border: 0;
  overflow: hidden;
  clip: rect(0 0 0 0);

  @media (min-width: 1280px) {
    display: inline-block;
    height: auto;
    margin: 0 0 0 0.5em;
    overflow: visible;
    position: relative;
    width: auto;
    clip: initial;
    font-size: 1rem;
  }
`;

export default Description;
