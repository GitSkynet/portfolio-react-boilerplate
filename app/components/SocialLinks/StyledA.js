import styled from 'styled-components';

const StyledA = styled.a`
  color: #465a65;
  font-size: 1.4rem;
  text-decoration: none;
`;

export default StyledA;
