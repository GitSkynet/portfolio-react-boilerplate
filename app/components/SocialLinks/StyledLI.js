import styled from 'styled-components';

const StyledLI = styled.li`
  padding: 0.5em;
  transition: transform 0.2s;
  font-size: 16px;

  &:hover {
    transform: scale(1.5);
  }

  @media (min-width: 37.5em) and (max-height: 31.25em) {
    /**/
  }
`;

export default StyledLI;
