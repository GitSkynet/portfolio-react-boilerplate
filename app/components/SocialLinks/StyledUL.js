import styled from 'styled-components';

const StyledUL = styled.ul`
  align-items: center;
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;

  & li:not(:first-child) {
    margin-left: 1rem;
  }
`;

export default StyledUL;
