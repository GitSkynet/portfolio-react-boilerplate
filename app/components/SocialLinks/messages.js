/*
 * SocialLinks Messages
 *
 * This contains all the text for the SocialLinks component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  header: {
    id: 'portfolio.social_links.header',
    defaultMessage: 'Also find me on:',
  },
});
