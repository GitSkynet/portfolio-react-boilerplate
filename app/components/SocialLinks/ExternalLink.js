import React from 'react';
import PropTypes from 'prop-types';
import FontAwesome from 'react-fontawesome';

import StyledA from './StyledA';

const ExternalLink = (props) => (
  <StyledA href={props.href} target="_blank" rel="noopener noreferrer">
    <FontAwesome name={props.faIcon} /><span hidden aria-hidden="false">{props.description}</span>
  </StyledA>
);

ExternalLink.propTypes = {
  description: PropTypes.string,
  href: PropTypes.string,
  faIcon: PropTypes.string,
};

ExternalLink.defaultProps = {
  faIcon: 'meh-o',
  description: 'ExternalLink to',
  href: '#',
};

export default ExternalLink;
