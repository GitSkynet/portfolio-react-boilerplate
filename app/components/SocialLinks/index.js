import React, { Component } from 'react';
import ReactTooltip from 'react-tooltip';
import { FormattedMessage } from 'react-intl';

import ExternalLink from './ExternalLink';
import Aside from './Aside';
import Header from './Header';
import StyledUL from './StyledUL';
import StyledLI from './StyledLI';
import messages from './messages';

class SocialLinks extends Component {
  componentDidMount() {
    if (this.list.children.length) {
      this.playAnimation(this.list.children);
    }
  }

  playAnimation = (links) => {
    const INDEX = 0; // starting node index
    const START_DELAY = 3000; // milliseconds;
    const DELAY = 100; // milliseconds;

    /* eslint no-param-reassign: "off" */
    setTimeout(function a(nodes, idx) {
      nodes[idx].style.transform = 'scale(1.5)';

      /* eslint consistent-return: "off" */
      setTimeout(() => {
        nodes[idx].style.removeProperty('transform');
        if (idx < nodes.length - 1) {
          setTimeout(a, DELAY, nodes, idx += 1);
        } else {
          return true;
        }
      }, DELAY);
    }, START_DELAY, links, INDEX);
  };

  render() {
    const externalLinks = [
      {
        icon: 'free-code-camp',
        name: 'freeCodeCamp',
        url: 'https://www.freecodecamp.org/mensae',
      },
      {
        icon: 'github',
        name: 'GitHub',
        url: 'https://github.com/Mensae',
      },
      {
        icon: 'linkedin-square',
        name: 'LinkedIn',
        url: 'https://nl.linkedin.com/in/jmeester',
      },
      {
        icon: 'codepen',
        name: 'Codepen',
        url: 'https://codepen.io/jmeester',
      },
    ];

    return (
      <Aside>
        <Header><FormattedMessage {...messages.header} /></Header>
        <StyledUL
          innerRef={
            (e) => {
              this.list = e;
            }
          }
        >
          {
            externalLinks.map((externalLink) => (
              <StyledLI
                key={externalLink.url}
                data-tip={externalLink.name}
              >
                <ExternalLink
                  href={externalLink.url}
                  faIcon={externalLink.icon}
                  description={externalLink.name}
                />
              </StyledLI>
            ))
          }
        </StyledUL>
        <ReactTooltip />
      </Aside>
    );
  }
}

export default SocialLinks;
