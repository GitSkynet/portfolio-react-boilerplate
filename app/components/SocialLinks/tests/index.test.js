import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { IntlProvider } from 'react-intl';

import SocialLinks from '../index';

import ThemeContext, { theme } from '../../../styles/theme';

describe('<SocialLinks />', () => {
  /**
   * TODO; Waiting for Enzyme to resolve issues with testing with the Context API
   * @link https://github.com/airbnb/enzyme/issues/1553
   */
  it.skip('should render and match the snapshot', () => {
    const wrapper = mount(
      <IntlProvider locale={'en'}>
        <ThemeContext.Provider value={theme}>
          <SocialLinks />
        </ThemeContext.Provider>
      </IntlProvider>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
