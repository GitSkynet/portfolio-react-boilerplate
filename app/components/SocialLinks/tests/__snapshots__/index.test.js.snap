// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks Should render and match the snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Aside__Aside>
    <Header__Header>
        <FormattedMessage
            defaultMessage="Also find me on:"
            id="portfolio.social_links.header"
            values={Object {}}
        />
    </Header__Header>
    <StyledUL__StyledUL
        innerRef={[Function]}
    >
        <StyledLI__StyledLI
            data-tip="freeCodeCamp"
        >
            <ExternalLink
                description="freeCodeCamp"
                faIcon="free-code-camp"
                href="https://www.freecodecamp.org/mensae"
            />
        </StyledLI__StyledLI>
        <StyledLI__StyledLI
            data-tip="GitHub"
        >
            <ExternalLink
                description="GitHub"
                faIcon="github"
                href="https://github.com/Mensae"
            />
        </StyledLI__StyledLI>
        <StyledLI__StyledLI
            data-tip="LinkedIn"
        >
            <ExternalLink
                description="LinkedIn"
                faIcon="linkedin-square"
                href="https://nl.linkedin.com/in/jmeester"
            />
        </StyledLI__StyledLI>
        <StyledLI__StyledLI
            data-tip="Codepen"
        >
            <ExternalLink
                description="Codepen"
                faIcon="codepen"
                href="https://codepen.io/jmeester"
            />
        </StyledLI__StyledLI>
    </StyledUL__StyledUL>
    <ReactTooltip
        insecure={true}
        resizeHide={true}
        wrapper="div"
    />
</Aside__Aside>,
  "nodes": Array [
    <Aside__Aside>
      <Header__Header>
            <FormattedMessage
                  defaultMessage="Also find me on:"
                  id="portfolio.social_links.header"
                  values={Object {}}
            />
      </Header__Header>
      <StyledUL__StyledUL
            innerRef={[Function]}
      >
            <StyledLI__StyledLI
                  data-tip="freeCodeCamp"
            >
                  <ExternalLink
                        description="freeCodeCamp"
                        faIcon="free-code-camp"
                        href="https://www.freecodecamp.org/mensae"
                  />
            </StyledLI__StyledLI>
            <StyledLI__StyledLI
                  data-tip="GitHub"
            >
                  <ExternalLink
                        description="GitHub"
                        faIcon="github"
                        href="https://github.com/Mensae"
                  />
            </StyledLI__StyledLI>
            <StyledLI__StyledLI
                  data-tip="LinkedIn"
            >
                  <ExternalLink
                        description="LinkedIn"
                        faIcon="linkedin-square"
                        href="https://nl.linkedin.com/in/jmeester"
                  />
            </StyledLI__StyledLI>
            <StyledLI__StyledLI
                  data-tip="Codepen"
            >
                  <ExternalLink
                        description="Codepen"
                        faIcon="codepen"
                        href="https://codepen.io/jmeester"
                  />
            </StyledLI__StyledLI>
      </StyledUL__StyledUL>
      <ReactTooltip
            insecure={true}
            resizeHide={true}
            wrapper="div"
      />
</Aside__Aside>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SocialLinks />,
      "_debugID": 344,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SocialLinks {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "playAnimation": [Function],
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Aside__Aside>
          <Header__Header>
                    <FormattedMessage
                              defaultMessage="Also find me on:"
                              id="portfolio.social_links.header"
                              values={Object {}}
                    />
          </Header__Header>
          <StyledUL__StyledUL
                    innerRef={[Function]}
          >
                    <StyledLI__StyledLI
                              data-tip="freeCodeCamp"
                    >
                              <ExternalLink
                                        description="freeCodeCamp"
                                        faIcon="free-code-camp"
                                        href="https://www.freecodecamp.org/mensae"
                              />
                    </StyledLI__StyledLI>
                    <StyledLI__StyledLI
                              data-tip="GitHub"
                    >
                              <ExternalLink
                                        description="GitHub"
                                        faIcon="github"
                                        href="https://github.com/Mensae"
                              />
                    </StyledLI__StyledLI>
                    <StyledLI__StyledLI
                              data-tip="LinkedIn"
                    >
                              <ExternalLink
                                        description="LinkedIn"
                                        faIcon="linkedin-square"
                                        href="https://nl.linkedin.com/in/jmeester"
                              />
                    </StyledLI__StyledLI>
                    <StyledLI__StyledLI
                              data-tip="Codepen"
                    >
                              <ExternalLink
                                        description="Codepen"
                                        faIcon="codepen"
                                        href="https://codepen.io/jmeester"
                              />
                    </StyledLI__StyledLI>
          </StyledUL__StyledUL>
          <ReactTooltip
                    insecure={true}
                    resizeHide={true}
                    wrapper="div"
          />
</Aside__Aside>,
        "_debugID": 345,
        "_renderedOutput": <Aside__Aside>
          <Header__Header>
                    <FormattedMessage
                              defaultMessage="Also find me on:"
                              id="portfolio.social_links.header"
                              values={Object {}}
                    />
          </Header__Header>
          <StyledUL__StyledUL
                    innerRef={[Function]}
          >
                    <StyledLI__StyledLI
                              data-tip="freeCodeCamp"
                    >
                              <ExternalLink
                                        description="freeCodeCamp"
                                        faIcon="free-code-camp"
                                        href="https://www.freecodecamp.org/mensae"
                              />
                    </StyledLI__StyledLI>
                    <StyledLI__StyledLI
                              data-tip="GitHub"
                    >
                              <ExternalLink
                                        description="GitHub"
                                        faIcon="github"
                                        href="https://github.com/Mensae"
                              />
                    </StyledLI__StyledLI>
                    <StyledLI__StyledLI
                              data-tip="LinkedIn"
                    >
                              <ExternalLink
                                        description="LinkedIn"
                                        faIcon="linkedin-square"
                                        href="https://nl.linkedin.com/in/jmeester"
                              />
                    </StyledLI__StyledLI>
                    <StyledLI__StyledLI
                              data-tip="Codepen"
                    >
                              <ExternalLink
                                        description="Codepen"
                                        faIcon="codepen"
                                        href="https://codepen.io/jmeester"
                              />
                    </StyledLI__StyledLI>
          </StyledUL__StyledUL>
          <ReactTooltip
                    insecure={true}
                    resizeHide={true}
                    wrapper="div"
          />
</Aside__Aside>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SocialLinks />,
}
`;
